src/components/ProbeStatus.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { FC, useState } from 'react';
+import { css } from 'emotion';
+import {
+  Badge,
+  BadgeColor,
+  Button,
+  Container,
+  ConfirmModal,
+  Legend,
+  IconName,
+  stylesFactory,
+  useTheme,
+} from '@grafana/ui';
+import { Probe, OrgRole } from 'types';
+import { SMDataSource } from 'datasource/DataSource';
+import { hasRole } from 'utils';
+import { UptimeGauge } from './UptimeGauge';
+import { GrafanaTheme } from '@grafana/data';
+
+interface Props {
+  probe: Probe;
+  instance: SMDataSource;
+  onResetToken: () => void;
+}
+
+interface BadgeStatus {
+  color: BadgeColor;
+  text: string;
+  icon: IconName;
+}
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  legend: css`
+    margin: ${theme.spacing.sm} ${theme.spacing.sm} 0 0;
+    width: auto;
+  `,
+  container: css`
+    padding-left: ${theme.spacing.sm}
+    margin-bottom: ${theme.spacing.md}
+  `,
+  badgeContainer: css`
+    margin-bottom: ${theme.spacing.sm};
+    display: flex;
+    align-items: center;
+  `,
+}));
+
+const getBadgeStatus = (online: boolean): BadgeStatus => {
+  if (online) {
+    return {
+      text: 'Online',
+      color: 'green',
+      icon: 'heart',
+    };
+  }
+  return {
+    text: 'Offline',
+    color: 'red',
+    icon: 'heart-break',
+  };
+};
+
+const ProbeStatus: FC<Props> = ({ probe, instance, onResetToken }) => {
+  const [showResetModal, setShowResetModal] = useState(false);
+  if (!probe) {
+    return null;
+  }
```

Comment:
Did React not complain about this? One of the hooks rules is violated here. I'd recommend installing the react-hooks eslint plugin (and import-order if you're into that sort of thing ;) ). See Alerting UI for how to make it work with grafana's base config.

Because it would throw:
```
React Hook "useTheme" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return? eslintreact-hooks/rules-of-hooks
```